### PR TITLE
fix(json): revert eager check

### PR DESCRIPTION
**Pull request by aarnphm**
This PR removes an eager check for a pydantic object.

Our JSONEncoder should already handle pydantic object lazily

https://bentoml.slack.com/archives/CKRANBHPH/p1660930201114169

### CODE REVIEW

**Review by yubozhao — APPROVED** [review]

**Review by yubozhao — APPROVED** [review]

File: bentoml/_internal/io_descriptors/json.py [hunk 1]
```diff
@@ -248,8 +248,15 @@ async def from_http_request(self, request: Request) -> JSONType:
     async def to_http_response(
         self, obj: JSONType | pydantic.BaseModel, ctx: Context | None = None
     ):
-        if isinstance(obj, pydantic.BaseModel):
-            obj = obj.dict()
+        try:
+            import pydantic
+
+            # This is to prevent cases where custom JSON encoder is used.
+            if isinstance(obj, pydantic.BaseModel):
```
Comment (parano) [inline bentoml/_internal/io_descriptors/json.py:255]: use `if self._pydantic_model`

Comment (aarnphm) [reply bentoml/_internal/io_descriptors/json.py:255]: This is not correct right? checking object instance is independent of whether `self._pydantic_model` is set or not.

File: bentoml/_internal/io_descriptors/json.py [hunk 2]
```diff
@@ -241,14 +241,21 @@ async def from_http_request(self, request: Request) -> JSONType:
...
 
     async def to_http_response(
         self, obj: JSONType | pydantic.BaseModel, ctx: Context | None = None
     ):
-        if isinstance(obj, pydantic.BaseModel):
+        if self._pydantic_model:
+            try:
+                obj = self._pydantic_model.validate(obj)
+            except pydantic.ValidationError as e:
+                raise BadInput(f"Invalid JSON input received: {e}") from e
```
Comment (aarnphm) [inline bentoml/_internal/io_descriptors/json.py:255]:
```suggestion
```

File: bentoml/_internal/io_descriptors/json.py [hunk 3]
```diff
@@ -241,14 +241,16 @@ async def from_http_request(self, request: Request) -> JSONType:
                 pydantic_model = self._pydantic_model.parse_obj(json_obj)
                 return pydantic_model
             except pydantic.ValidationError as e:
-                raise BadInput(f"Invalid JSON input received: {e}") from None
+                raise BadInput(f"Invalid JSON input received: {e}") from e
```
Comment (sauyon) [inline bentoml/_internal/io_descriptors/json.py:244]: Hm, do you know why this was `None` before?

**Review by sauyon — APPROVED** [review]
LGTM 👍

File: bentoml/_internal/io_descriptors/json.py (hunk 3 above)
Comment (aarnphm) [reply bentoml/_internal/io_descriptors/json.py:244]: no idea.

Comment (parano) [reply bentoml/_internal/io_descriptors/json.py:244]: @sauyon @aarnphm https://peps.python.org/pep-0415/ that was for better exception context

**Pull request merged**